src/test/java/net/snowflake/client/jdbc/ProxyLatestIT.java

```diff
@@ -0,0 +1,401 @@
+        .alias("wait for wiremock responding")
+        .atMost(Duration.ofSeconds(10))
+        .until(
+            () -> {
+              try {
+                httpProxyPort = findFreePort();
+                httpsProxyPort = findFreePort();
+                wiremockStandalone =
+                    new ProcessBuilder(
+                            "java",
+                            "-jar",
+                            getWiremockStandAlonePath(),
+                            "--root-dir",
+                            System.getProperty("user.dir")
+                                + File.separator
+                                + WIREMOCK_HOME_DIR
+                                + File.separator,
+                            "--enable-browser-proxying", // work as forward proxy
+                            "--proxy-pass-through",
+                            "false", // pass through only matched requests
+                            "--port",
+                            String.valueOf(httpProxyPort),
+                            "--https-port",
+                            String.valueOf(httpsProxyPort),
+                            "--https-keystore",
+                            getResourceURL("wiremock" + File.separator + "ca-cert.jks"),
+                            "--ca-keystore",
+                            getResourceURL("wiremock" + File.separator + "ca-cert.jks"))
+                        .inheritIO()
+                        .start();
+                waitForWiremock();
+                return true;
+              } catch (Exception e) {
+                logger.warning("Failed to start wiremock, retrying: " + e);
+                return false;
+              }
+            });
+  }
+
+  private void resetWiremock() {
+    HttpPost postRequest;
+    postRequest = new HttpPost("http://" + WIREMOCK_HOST + ":" + getAdminPort() + "/__admin/reset");
+    try (CloseableHttpClient client = HttpClients.createDefault();
+        CloseableHttpResponse response = client.execute(postRequest)) {
+      assertEquals(200, response.getStatusLine().getStatusCode());
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static String getWiremockStandAlonePath() {
+    return System.getProperty("user.home") + WIREMOCK_M2_PATH;
+  }
+
+  private static void waitForWiremock() {
+    await()
+        .pollDelay(Duration.ofSeconds(1))
+        .atMost(Duration.ofSeconds(3))
+        .until(ProxyLatestIT::isWiremockResponding);
+  }
+
+  private static boolean isWiremockResponding() {
+    try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+      HttpGet request =
+          new HttpGet(String.format("http://%s:%d/__admin/mappings", WIREMOCK_HOST, httpProxyPort));
+      CloseableHttpResponse response = httpClient.execute(request);
+      return response.getStatusLine().getStatusCode() == 200;
+    } catch (Exception e) {
+      logger.warning("Waiting for wiremock to respond: " + e);
```

Comment:
exception should be passed as a separate argument - it will be possible when you use SfLogger of Slf4j

Reply:
done